conda/core/initialize.py

```diff
@@ -1504,7 +1509,9 @@ def _powershell_profile_content(conda_prefix):
     conda_powershell_module = dals("""
     #region conda initialize
     # !! Contents within this block are managed by 'conda init' !!
-    (& "{conda_exe}" "shell.powershell" "hook") | Out-String | Invoke-Expression
+    If (Test-Path "{conda_exe}") {{
+        (& "{conda_exe}" "shell.powershell" "hook") | Out-String | ?{{$_}} | Invoke-Expression
```

Comment:
@kenodegard : we have had an issue today where `Test-Path "{conda_exe}"` still worked as `conda.exe` existed, but Python couldn't be executed due to some missing files/DLLs (it was a broken installation).

However this led to other scripts failing during PS1 initialization, as the line above then failed: conda was there, and could be called, but Python crashed → the whole profile script was treated as failed.

Should I add an extra layer of safety here and catch exceptions occurring during this conda hook call?

Reply:
At the moment, I am using this code:

```powershell
#region conda initialize
# !! Contents within this block are managed by 'conda init' !!
If (Test-Path "C:\Program Files\Anaconda3\Scripts\conda.exe") {
    try {
        (& "C:\Program Files\Anaconda3\Scripts\conda.exe" "shell.powershell" "hook") | Out-String | Invoke-Expression
    }
    catch {
        # some verbose warning message which tells the user
        # what to do, i.e. to reinstall Anaconda.
    }
}
#endregion
```